<a href="https://colab.research.google.com/github/Devaprakash917/SCT_DT_02/blob/main/Data_Cleaning_and_Preparation.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

import necessary library

In [1]:
import pandas as pd

read csv file and store the file in df variable

In [2]:
df=pd.read_csv('/content/world_development_measurement.csv')

In [3]:
df.head(5)

,Birth Rate,Business Tax Rate,CO2 Emissions,Country,Days to Start Business,Ease of Business,Energy Usage,GDP,Health Exp % GDP,Health Exp/Capita,...,Life Expectancy Male,Mobile Phone Usage,Number of Records,Population 0-14,Population 15-64,Population 65+,Population Total,Population Urban,Tourism Inbound,Tourism Outbound
0,0.020,NaN,87931.0,Algeria,NaN,NaN,26998.0,"$54,790,058,957",0.035,$60,...,67.0,0.0,1,0.342,0.619,0.039,31719449,0.599,"$102,000,000","$193,000,000"
1,0.050,NaN,9542.0,Angola,NaN,NaN,7499.0,"$9,129,594,819",0.034,$22,...,44.0,0.0,1,0.476,0.499,0.025,13924930,0.324,"$34,000,000","$146,000,000"
2,0.043,NaN,1617.0,Benin,NaN,NaN,1983.0,"$2,359,122,303",0.043,$15,...,53.0,0.0,1,0.454,0.517,0.029,6949366,0.383,"$77,000,000","$50,000,000"
3,0.027,NaN,4276.0,Botswana,NaN,NaN,1836.0,"$5,788,311,645",0.047,$152,...,49.0,0.1,1,0.383,0.587,0.029,1755375,0.532,"$227,000,000","$209,000,000"
4,0.046,NaN,1041.0,Burkina Faso,NaN,NaN,NaN,"$2,610,959,139",0.051,$12,...,49.0,0.0,1,0.468,0.505,0.028,11607944,0.178,"$23,000,000","$30,000,000"


The df.head(5) command is used to display the first five rows of the dataset, helping to quickly understand the structure, columns, and identify missing values (NaN) for further data cleaning.

In [4]:
df.tail(5)  #It displays last 5 rows of dataset

,Birth Rate,Business Tax Rate,CO2 Emissions,Country,Days to Start Business,Ease of Business,Energy Usage,GDP,Health Exp % GDP,Health Exp/Capita,...,Life Expectancy Male,Mobile Phone Usage,Number of Records,Population 0-14,Population 15-64,Population 65+,Population Total,Population Urban,Tourism Inbound,Tourism Outbound
2699,NaN,NaN,NaN,Turks and Caicos Islands,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,1,NaN,NaN,NaN,32427,0.911,NaN,NaN
2700,0.013,46.4%,NaN,United States,5.0,4.0,2132446.0,"$16,244,600,000,000",0.179,"$8,895",...,76.0,1.0,1,0.196,0.667,0.136,313873685,0.811,"$200,092,000,000","$126,573,000,000"
2701,0.015,41.9%,NaN,Uruguay,7.0,85.0,NaN,"$50,004,354,667",0.089,"$1,308",...,74.0,1.5,1,0.220,0.639,0.140,3395253,0.948,"$2,222,000,000","$1,028,000,000"
2702,0.020,61.9%,NaN,"Venezuela, RB",144.0,180.0,NaN,"$381,286,223,859",0.046,$593,...,72.0,1.0,1,0.288,0.652,0.060,29954782,0.889,"$904,000,000","$3,202,000,000"
2703,0.011,NaN,NaN,Virgin Islands (U.S.),NaN,NaN,NaN,NaN,NaN,NaN,...,76.0,NaN,1,0.206,0.641,0.152,105275,0.949,NaN,NaN


The df.tail(5) command is used to display the last five rows of the dataset, helping to quickly understand the structure, columns, and identify missing values (NaN) for further data cleaning.

In [5]:
df['Number of Records'].value_counts()

,count
Number of Records,
1,2704


It is used to know how many unique values are present in a single column

In [6]:
df.duplicated().sum()

np.int64(0)

The df.duplicated().sum() function is used to identify duplicate rows in the dataset, and the result shows 0, indicating no duplicate records are present.

In [7]:
df.shape

(2704, 25)

The df.shape function returns the dimensions of the dataset, showing that it contains 2704 rows and 25 columns.

Statastical before filling nan is for understanding the dataset

In [8]:
df.describe(include='all')

,Birth Rate,Business Tax Rate,CO2 Emissions,Country,Days to Start Business,Ease of Business,Energy Usage,GDP,Health Exp % GDP,Health Exp/Capita,...,Life Expectancy Male,Mobile Phone Usage,Number of Records,Population 0-14,Population 15-64,Population 65+,Population Total,Population Urban,Tourism Inbound,Tourism Outbound
count,2585.000000,1423,2.125000e+03,2704,1718.000000,185.000000,1.785000e+03,2494,2395.000000,2395,...,2568.000000,2537.000000,2704.0,2484.000000,2484.000000,2484.000000,2.704000e+03,2678.000000,2336,2233
unique,NaN,447,NaN,208,NaN,NaN,NaN,2494,NaN,1069,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1650,1457
top,NaN,44.1%,NaN,Algeria,NaN,NaN,NaN,"$381,286,223,859",NaN,$15,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,"$46,000,000","$8,000,000"
freq,NaN,20,NaN,13,NaN,NaN,NaN,1,NaN,27,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,9,19
mean,0.022715,NaN,1.423071e+05,NaN,39.999418,94.875676,7.723684e+04,NaN,0.064170,NaN,...,66.461838,0.578124,1.0,0.303051,0.623481,0.073465,3.145729e+07,0.562953,NaN,NaN
std,0.011338,NaN,6.159288e+05,NaN,57.318588,54.791906,2.599239e+05,NaN,0.026325,NaN,...,9.390217,0.474647,0.0,0.106577,0.069660,0.049541,1.242894e+08,0.245897,NaN,NaN
min,0.007000,NaN,7.000000e+00,NaN,1.000000,1.000000,8.000000e+00,NaN,0.008000,NaN,...,37.000000,0.000000,1.0,0.118000,0.474000,0.003000,1.887600e+04,0.082000,NaN,NaN
25%,0.013000,NaN,1.360000e+03,NaN,13.000000,48.000000,3.737000e+03,NaN,0.046000,NaN,...,61.000000,0.100000,1.0,0.204750,0.559000,0.033000,8.783360e+05,0.356000,NaN,NaN
50%,0.020000,NaN,8.529000e+03,NaN,26.000000,94.000000,1.433800e+04,NaN,0.061000,NaN,...,69.000000,0.500000,1.0,0.299500,0.642500,0.052000,5.800324e+06,0.560000,NaN,NaN
75%,0.031000,NaN,5.910800e+04,NaN,45.000000,142.000000,4.185200e+04,NaN,0.079000,NaN,...,73.250000,0.900000,1.0,0.402000,0.676000,0.112000,2.014089e+07,0.765000,NaN,NaN


It provides summary statistics such as count, mean, standard deviation, minimum, maximum, and percentiles for numerical columns, along with unique values and frequency for categorical columns. This helps in understanding the data distribution and identifying missing values (NaN) before further data cleaning.

In [9]:
df.skew(numeric_only=True)

,0
Birth Rate,0.662151
CO2 Emissions,8.654039
Days to Start Business,7.437592
Ease of Business,0.017745
Energy Usage,7.062230
Health Exp % GDP,1.179757
Hours to do Tax,4.491308
Infant Mortality Rate,1.049118
Internet Usage,0.940966
Lending Interest,15.536119


Skewness indicates whether the data is symmetrically distributed or biased toward one side. Positive values represent right-skewed distributions, while negative values indicate left-skewed data. This helps in understanding data distribution and deciding if transformation is needed before analysis or modeling.

In [10]:
df.var(numeric_only=True)

,0
Birth Rate,1.285599e-04
CO2 Emissions,3.793683e+11
Days to Start Business,3.285421e+03
Ease of Business,3.002153e+03
Energy Usage,6.756041e+10
Health Exp % GDP,6.929846e-04
Hours to do Tax,7.469272e+04
Infant Mortality Rate,8.980571e-04
Internet Usage,6.766855e-02
Lending Interest,2.952523e-02


Variance measures how much the data points are spread out from the mean. Higher values indicate greater variability, while lower values suggest the data is more consistent. This helps in understanding the dispersion of each feature before further analysis.

In [11]:
df.std(numeric_only=True)

,0
Birth Rate,1.133843e-02
CO2 Emissions,6.159288e+05
Days to Start Business,5.731859e+01
Ease of Business,5.479191e+01
Energy Usage,2.599239e+05
Health Exp % GDP,2.632460e-02
Hours to do Tax,2.732997e+02
Infant Mortality Rate,2.996760e-02
Internet Usage,2.601318e-01
Lending Interest,1.718291e-01


Standard deviation measures the spread of data around the mean in the same units as the original data. Higher values indicate greater variability, while lower values show that the data points are closer to the mean.

In [12]:
df.corr(numeric_only=True)

,Birth Rate,CO2 Emissions,Days to Start Business,Ease of Business,Energy Usage,Health Exp % GDP,Hours to do Tax,Infant Mortality Rate,Internet Usage,Lending Interest,Life Expectancy Female,Life Expectancy Male,Mobile Phone Usage,Number of Records,Population 0-14,Population 15-64,Population 65+,Population Total,Population Urban
Birth Rate,1.000000,-0.175750,0.169013,0.727492,-0.169664,-0.227653,0.069324,0.872535,-0.656132,0.281503,-0.868174,-0.825745,-0.558411,NaN,0.955966,-0.912837,-0.773014,-0.057851,-0.606933
CO2 Emissions,-0.175750,1.000000,-0.055666,NaN,0.988715,0.163236,0.081485,-0.135421,0.137445,-0.085876,0.135219,0.141452,0.054400,NaN,-0.198328,0.190808,0.160403,0.721145,0.098051
Days to Start Business,0.169013,-0.055666,1.000000,0.308789,-0.069418,-0.147686,0.148423,0.200082,-0.243011,0.146216,-0.197764,-0.213764,-0.199868,NaN,0.203978,-0.166224,-0.201973,0.001357,-0.092493
Ease of Business,0.727492,NaN,0.308789,1.000000,-0.234867,-0.278061,0.346213,0.731639,-0.774005,0.488395,-0.709671,-0.706122,-0.582954,NaN,0.741035,-0.631452,-0.646456,0.029175,-0.493255
Energy Usage,-0.169664,0.988715,-0.069418,-0.234867,1.000000,0.253146,0.074673,-0.120019,0.154787,-0.085713,0.125224,0.132984,0.045198,NaN,-0.184007,0.171528,0.148971,0.677971,0.094218
Health Exp % GDP,-0.227653,0.163236,-0.147686,-0.278061,0.253146,1.000000,-0.079755,-0.165117,0.333544,-0.028021,0.208335,0.219441,0.187336,NaN,-0.292266,0.093604,0.489835,-0.054097,0.205994
Hours to do Tax,0.069324,0.081485,0.148423,0.346213,0.074673,-0.079755,1.000000,0.133808,-0.200398,0.140826,-0.104772,-0.154317,-0.103461,NaN,0.075677,-0.084376,-0.043624,0.146920,0.010022
Infant Mortality Rate,0.872535,-0.135421,0.200082,0.731639,-0.120019,-0.165117,0.133808,1.000000,-0.639990,0.304399,-0.933001,-0.912240,-0.599383,NaN,0.822479,-0.798960,-0.649284,0.006897,-0.619742
Internet Usage,-0.656132,0.137445,-0.243011,-0.774005,0.154787,0.333544,-0.200398,-0.639990,1.000000,-0.274177,0.667712,0.686260,0.754287,NaN,-0.731942,0.626876,0.689699,-0.045266,0.567800
Lending Interest,0.281503,-0.085876,0.146216,0.488395,-0.085713,-0.028021,0.140826,0.304399,-0.274177,1.000000,-0.340463,-0.350880,-0.269733,NaN,0.278345,-0.277096,-0.208888,-0.040289,-0.170459


It shows how strongly each variable is related to others, with values ranging from -1 to +1. Positive values indicate a direct relationship, negative values indicate an inverse relationship, and values close to 0 suggest little or no correlation. This helps in identifying relationships between variables for analysis and modeling.

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2704 entries, 0 to 2703
Data columns (total 25 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Birth Rate              2585 non-null   float64
 1   Business Tax Rate       1423 non-null   object 
 2   CO2 Emissions           2125 non-null   float64
 3   Country                 2704 non-null   object 
 4   Days to Start Business  1718 non-null   float64
 5   Ease of Business        185 non-null    float64
 6   Energy Usage            1785 non-null   float64
 7   GDP                     2494 non-null   object 
 8   Health Exp % GDP        2395 non-null   float64
 9   Health Exp/Capita       2395 non-null   object 
 10  Hours to do Tax         1416 non-null   float64
 11  Infant Mortality Rate   2444 non-null   float64
 12  Internet Usage          2531 non-null   float64
 13  Lending Interest        1880 non-null   float64
 14  Life Expectancy Female  2568 non-null   

It provides details such as the total number of rows and columns, column names, non-null counts, and data types of each column. This helps in identifying missing values and understanding the structure of the dataset before performing data cleaning.

In [14]:
df.isnull().sum()

,0
Birth Rate,119
Business Tax Rate,1281
CO2 Emissions,579
Country,0
Days to Start Business,986
Ease of Business,2519
Energy Usage,919
GDP,210
Health Exp % GDP,309
Health Exp/Capita,309


It helps identify which features have incomplete data and the extent of missing values, making it easier to decide appropriate data cleaning techniques such as imputation or removal.

In [15]:
threshold=len(df)*0.6
df=df.dropna(axis=1,thresh=threshold)

A threshold is set at 60% of the total rows, and dropna(axis=1, thresh=threshold) removes columns that do not meet this minimum number of non-null values.

In [16]:
len(df.columns)

22

 The len(df.columns) function then shows the updated number of columns remaining after cleaning.

In [17]:
currency_cols=['GDP','Health Exp/Capita','Tourism Inbound','Tourism Outbound']

for col in currency_cols:
  df[col]=df[col].str.replace(',','').str.replace('$','').astype(float)

First, selected currency-related columns (like GDP and Health Expenditure per Capita) are converted from string format to numeric by removing commas and dollar signs, and then casting them to float.

In [18]:
numeric_cols=[]
categorical_cols=[]
for col in df.columns:
  if df[col].dtype=='object':
    categorical_cols.append(col)
  else:
    numeric_cols.append(col)
print('Numeric Columns:',numeric_cols)
print('Categorical Columns:',categorical_cols)


Numeric Columns: ['Birth Rate', 'CO2 Emissions', 'Days to Start Business', 'Energy Usage', 'GDP', 'Health Exp % GDP', 'Health Exp/Capita', 'Infant Mortality Rate', 'Internet Usage', 'Lending Interest', 'Life Expectancy Female', 'Life Expectancy Male', 'Mobile Phone Usage', 'Number of Records', 'Population 0-14', 'Population 15-64', 'Population 65+', 'Population Total', 'Population Urban', 'Tourism Inbound', 'Tourism Outbound']
Categorical Columns: ['Country']


 Next, the dataset columns are separated into numerical and categorical features based on their data types, which helps in applying appropriate analysis and preprocessing techniques.

In [19]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2704 entries, 0 to 2703
Data columns (total 22 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Birth Rate              2585 non-null   float64
 1   CO2 Emissions           2125 non-null   float64
 2   Country                 2704 non-null   object 
 3   Days to Start Business  1718 non-null   float64
 4   Energy Usage            1785 non-null   float64
 5   GDP                     2494 non-null   float64
 6   Health Exp % GDP        2395 non-null   float64
 7   Health Exp/Capita       2395 non-null   float64
 8   Infant Mortality Rate   2444 non-null   float64
 9   Internet Usage          2531 non-null   float64
 10  Lending Interest        1880 non-null   float64
 11  Life Expectancy Female  2568 non-null   float64
 12  Life Expectancy Male    2568 non-null   float64
 13  Mobile Phone Usage      2537 non-null   float64
 14  Number of Records       2704 non-null   

Now columns are of desired data type

In [20]:
for col in numeric_cols:
  df[col]=df[col].fillna(df[col].mean())

It iterates through each numeric column and replaces null values with the mean of that column using fillna(). This ensures that missing data is imputed without significantly affecting the overall distribution of the dataset.

In [21]:
df

,Birth Rate,CO2 Emissions,Country,Days to Start Business,Energy Usage,GDP,Health Exp % GDP,Health Exp/Capita,Infant Mortality Rate,Internet Usage,...,Life Expectancy Male,Mobile Phone Usage,Number of Records,Population 0-14,Population 15-64,Population 65+,Population Total,Population Urban,Tourism Inbound,Tourism Outbound
0,0.020000,87931.000000,Algeria,39.999418,2.699800e+04,5.479006e+10,0.03500,60.000000,0.034000,0.000000,...,67.000000,0.000000,1,0.342000,0.619000,0.039000,31719449,0.599,1.020000e+08,1.930000e+08
1,0.050000,9542.000000,Angola,39.999418,7.499000e+03,9.129595e+09,0.03400,22.000000,0.128000,0.000000,...,44.000000,0.000000,1,0.476000,0.499000,0.025000,13924930,0.324,3.400000e+07,1.460000e+08
2,0.043000,1617.000000,Benin,39.999418,1.983000e+03,2.359122e+09,0.04300,15.000000,0.090000,0.000000,...,53.000000,0.000000,1,0.454000,0.517000,0.029000,6949366,0.383,7.700000e+07,5.000000e+07
3,0.027000,4276.000000,Botswana,39.999418,1.836000e+03,5.788312e+09,0.04700,152.000000,0.054000,0.000000,...,49.000000,0.100000,1,0.383000,0.587000,0.029000,1755375,0.532,2.270000e+08,2.090000e+08
4,0.046000,1041.000000,Burkina Faso,39.999418,7.723684e+04,2.610959e+09,0.05100,12.000000,0.096000,0.000000,...,49.000000,0.000000,1,0.468000,0.505000,0.028000,11607944,0.178,2.300000e+07,3.000000e+07
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2699,0.022715,142307.107765,Turks and Caicos Islands,39.999418,7.723684e+04,2.638540e+11,0.06417,801.397077,0.032813,0.239747,...,66.461838,0.578124,1,0.303051,0.623481,0.073465,32427,0.911,4.939320e+09,4.729435e+09
2700,0.013000,142307.107765,United States,5.000000,2.132446e+06,1.624460e+13,0.17900,8895.000000,0.006000,0.800000,...,76.000000,1.000000,1,0.196000,0.667000,0.136000,313873685,0.811,2.000920e+11,1.265730e+11
2701,0.015000,142307.107765,Uruguay,7.000000,7.723684e+04,5.000435e+10,0.08900,1308.000000,0.010000,0.500000,...,74.000000,1.500000,1,0.220000,0.639000,0.140000,3395253,0.948,2.222000e+09,1.028000e+09
2702,0.020000,142307.107765,"Venezuela, RB",144.000000,7.723684e+04,3.812862e+11,0.04600,593.000000,0.013000,0.500000,...,72.000000,1.000000,1,0.288000,0.652000,0.060000,29954782,0.889,9.040000e+08,3.202000e+09


store the cleaned csv file so it can be used for model building

In [22]:
df.to_csv("cleaned_world_development_measurement.csv",index=False)

In [23]:
from google.colab import files
files.download('cleaned_world_development_measurement.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

Validation

In [24]:
cleaned_df=pd.read_csv('/content/cleaned_world_development_measurement.csv')

In [25]:
cleaned_df.isnull().sum()

,0
Birth Rate,0
CO2 Emissions,0
Country,0
Days to Start Business,0
Energy Usage,0
GDP,0
Health Exp % GDP,0
Health Exp/Capita,0
Infant Mortality Rate,0
Internet Usage,0


In [26]:
cleaned_df.duplicated().sum()

np.int64(0)

In [27]:
cleaned_df.dtypes

,0
Birth Rate,float64
CO2 Emissions,float64
Country,object
Days to Start Business,float64
Energy Usage,float64
GDP,float64
Health Exp % GDP,float64
Health Exp/Capita,float64
Infant Mortality Rate,float64
Internet Usage,float64


Statastical Analysis After filling nan values is to validate dataset

In [28]:
cleaned_df.describe(include='all')

,Birth Rate,CO2 Emissions,Country,Days to Start Business,Energy Usage,GDP,Health Exp % GDP,Health Exp/Capita,Infant Mortality Rate,Internet Usage,...,Life Expectancy Male,Mobile Phone Usage,Number of Records,Population 0-14,Population 15-64,Population 65+,Population Total,Population Urban,Tourism Inbound,Tourism Outbound
count,2704.000000,2.704000e+03,2704,2704.000000,2.704000e+03,2.704000e+03,2704.000000,2704.000000,2704.000000,2704.000000,...,2704.000000,2704.000000,2704.0,2704.000000,2704.000000,2704.000000,2.704000e+03,2704.000000,2.704000e+03,2.704000e+03
unique,NaN,NaN,208,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
top,NaN,NaN,Algeria,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
freq,NaN,NaN,13,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
mean,0.022715,1.423071e+05,NaN,39.999418,7.723684e+04,2.638540e+11,0.064170,801.397077,0.032813,0.239747,...,66.461838,0.578124,1.0,0.303051,0.623481,0.073465,3.145729e+07,0.562953,4.939320e+09,4.729435e+09
std,0.011086,5.459901e+05,NaN,45.683314,2.111643e+05,1.085131e+12,0.024774,1386.672409,0.028490,0.251670,...,9.150937,0.459751,0.0,0.102148,0.066765,0.047482,1.242894e+08,0.244712,1.287363e+10,1.205649e+10
min,0.007000,7.000000e+00,NaN,1.000000,8.000000e+00,6.310127e+07,0.008000,2.000000,0.002000,0.000000,...,37.000000,0.000000,1.0,0.118000,0.474000,0.003000,1.887600e+04,0.082000,7.000000e+05,2.000000e+05
25%,0.013000,2.237000e+03,NaN,20.000000,7.851750e+03,4.113748e+09,0.048000,58.000000,0.010000,0.000000,...,61.000000,0.200000,1.0,0.210000,0.565000,0.034000,8.783360e+05,0.357750,1.860000e+08,1.470000e+08
50%,0.021000,2.990450e+04,NaN,39.999418,4.485050e+04,1.972267e+10,0.064000,262.500000,0.025000,0.200000,...,68.000000,0.578124,1.0,0.303051,0.631000,0.059000,5.800324e+06,0.562953,1.103500e+09,8.255000e+08
75%,0.030000,1.423071e+05,NaN,39.999418,7.723684e+04,1.890042e+11,0.076000,801.397077,0.049000,0.400000,...,73.000000,0.900000,1.0,0.395000,0.673000,0.105000,2.014089e+07,0.763000,4.939320e+09,4.729435e+09


In [29]:
cleaned_df.skew(numeric_only=True)

,0
Birth Rate,0.677203
CO2 Emissions,9.760611
Days to Start Business,9.327939
Energy Usage,8.689642
GDP,9.812540
Health Exp % GDP,1.253465
Health Exp/Capita,2.993509
Infant Mortality Rate,1.103447
Internet Usage,0.972557
Lending Interest,18.627787


In [30]:
cleaned_df.var(numeric_only=True)

,0
Birth Rate,1.229001e-04
CO2 Emissions,2.981052e+11
Days to Start Business,2.086965e+03
Energy Usage,4.459037e+10
GDP,1.177509e+24
Health Exp % GDP,6.137644e-04
Health Exp/Capita,1.922860e+06
Infant Mortality Rate,8.116735e-04
Internet Usage,6.333756e-02
Lending Interest,2.052457e-02


In [31]:
cleaned_df.std(numeric_only=True)

,0
Birth Rate,1.108603e-02
CO2 Emissions,5.459901e+05
Days to Start Business,4.568331e+01
Energy Usage,2.111643e+05
GDP,1.085131e+12
Health Exp % GDP,2.477427e-02
Health Exp/Capita,1.386672e+03
Infant Mortality Rate,2.848988e-02
Internet Usage,2.516696e-01
Lending Interest,1.432640e-01
